File: streamer-v2/src/main/java/eu/tsalliance/streamer/console/Command.java
package eu.tsalliance.streamer.console;

import lombok.Getter;
import org.jetbrains.annotations.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.ArrayList;
import java.util.List;

public abstract class Command {
    private static final Logger logger = LoggerFactory.getLogger(CommandHandler.class);

    @Getter private final String name;
    @Getter private final String usage;
    @Getter private final String description;
    @Getter private final List<String> aliases;

    public Command(@NotNull String name, String usage, String description, @NotNull List<String> aliases) {
        this.name = name;
        this.description = description;
        this.usage = usage;
        this.aliases = aliases;
    }

    public abstract void execute(String name, ArrayList<String> args);

    public void sendUsage() {
        logger.warn("Usage: " + this.getName() + " " + this.getUsage());
    }

}

File: master/src/main/kotlin/live/tsradio/master/console/Command.kt
package live.tsradio.master.console

import org.slf4j.Logger
import org.slf4j.LoggerFactory

abstract class Command(val name: String, val usage: String, val description: String) {
    private val logger: Logger = LoggerFactory.getLogger(CommandHandler::class.java)

    abstract fun execute(name: String, args: ArrayList<String>)

    fun sendUsage(){
        logger.warn("Did you mean: '$name $usage' ?")
    }
    fun sendText(text: String){
        logger.info(text)
    }
}